### PR TITLE
Np 6194 open person endpoint

### PR DESCRIPTION
**Pull request by svenngjeto**
Verifying access to read NIN by reading optional Authorization header

### CODE REVIEW

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/userAdministrator.feature [hunk 1]
```diff
@@ -33,20 +33,18 @@ Feature: API tests for Cristin Person fetch
...
+    Given path '/person/' + samplePersonIdentifier
+    When method GET
+    Then status 200
+    And response.NationalIdentificationNumber != '#present'
 
-  Scenario: Fetch returns 401 Unauthorized when not authenticated
+  Scenario: Fetch returns unclassified person data when not authenticated nor authorized with strange token
     Given path '/person/' + samplePersonIdentifier
-    And request
+    * header Authorization = 'Bearer and.just-a.silly-text-for-token'
```
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/userAdministrator.feature:44]: Should we return Forbidden or OK when user sends token? OK without token is OK but should we explicitly deny a user access when with an invalid token?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/userAdministrator.feature [hunk 2]
```diff
@@ -33,20 +33,18 @@ Feature: API tests for Cristin Person fetch
     Then status 200
     And response.NationalIdentificationNumber != '#present'
 
-  Scenario: Fetch returns status 403 Forbidden when not authorized
```
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/userAdministrator.feature:36]: Why are you deleting this?

**Review by torarnet — COMMENTED** [review]
Not skilled enough in authorization logic to do a good PR, leaving a comment instead.

File: cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java [hunk 3]
```diff
@@ -72,4 +84,20 @@ private static boolean clientIsInternalBackend(RequestInfo requestInfo) {
...
+    private static boolean requesterHasAccessRightInBearerToken(RequestInfo requestInfo) {
+        try {
+            final var authorizationHeader = requestInfo.getHeaders().get(AUTHORIZATION);
+            if (nonNull(authorizationHeader) && authorizationHeader.startsWith(BEARER)) {
+                var token = authorizationHeader.substring(BEARER.length() + 1, authorizationHeader.length());
+                var payload = new String(Base64.getUrlDecoder().decode(token.split(DOT_SEPARATOR)[PAYLOAD_SEGMENT]));
+                var accessRights = JsonUtils.dtoObjectMapper.readTree(new StringReader(payload))
+                        .get(CUSTOM_ACCESS_RIGHTS).asText();
+                return accessRights.contains(EDIT_OWN_INSTITUTION_USERS);
+            }
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java:97]: Can you use https://github.com/auth0/java-jwt to *verify* and extract the relevant parts?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java [hunk 4]
```diff
@@ -43,33 +58,60 @@ public static void validateIdentificationNumberAccess(RequestInfo requestInfo) t
...
+                .filter(BACKEND_SCOPE_AS_DEFINED_IN_IDENTITY_SERVICE::equals)
+                .isPresent();
     }
 
     private static boolean requesterHasNoAccessRightToEditProjects(RequestInfo requestInfo) {
         return !requestInfo.getAccessRights().contains(EDIT_OWN_INSTITUTION_PROJECTS);
     }
+
+    private static boolean requesterHasAccessRightInBearerToken(RequestInfo requestInfo) throws UnauthorizedException {
+        final var authorizationHeader = requestInfo.getHeaders().get(AUTHORIZATION);
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java:93]: Are we mixing in `var` for a reason?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: docs/cristin-proxy-swagger.yaml [hunk 5]
```diff
@@ -469,6 +467,8 @@ paths:
                   $ref: '#/components/examples/PersonExample'
         400:
           $ref: '#/components/responses/400'
+        401:
+          $ref: '#/components/responses/400'
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply docs/cristin-proxy-swagger.yaml]: 400:=401...

File: cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java (hunk 3 above)
Comment (svenngjeto) [reply cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java]: Verifying token with keys from Cognito

**Review by brinxmat — APPROVED** [review]

**Pull request merged**